src/test/java/com/facebook/presto/hadoop/TestLineReader.java

```diff
@@ -0,0 +1,173 @@
+package com.facebook.presto.hadoop;
+
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.util.LineReader;
+import org.testng.annotations.Test;
+
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.InputStream;
+import java.util.Arrays;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
+import static org.testng.Assert.assertEquals;
+import static org.testng.Assert.fail;
+
+public class TestLineReader
+{
+    @Test
+    public void testDefaultReaderZeroBytes()
+    {
+        byte[] input = "Hello world! Goodbye world!\n".getBytes(UTF_8);
+        InputStream in = new ByteArrayInputStream(input);
+        // Set the LineReader internal read buffer size 4 bytes
+        LineReader reader = new LineReader(in, 4);
+        Text str = new Text();
+        try {
+            // There should not be any bytes read into str because the maxLineLength is 0
+            reader.readLine(str, 0, 30);
+            assertEquals(str.getLength(), 0);
+            assertEquals(str.getBytes(), "".getBytes());
+        }
+        catch (IOException e) {
+            fail(e.getMessage());
+        }
+    }
+
+    @Test
+    public void testDefaultReaderNonZeroBytes()
+    {
+        byte[] input = "Hello world! Goodbye world!\n".getBytes(UTF_8);
+        InputStream in = new ByteArrayInputStream(input);
+        // Set the LineReader internal read buffer size 4 bytes
+        LineReader reader = new LineReader(in, 4);
+        Text str = new Text();
+        try {
+            reader.readLine(str, 30, 30);
+            // The LineReader does not store the new line character into str,
+            // and the str was doubled 3 times so its size is 32, with 5 0-valued
+            // bytes at the end, so we need to compare just the first 27 characters.
+            assertEquals(str.getLength(), 27);
```

Comment:
I would write this as
```java
String input = "Hello world! Goodbye world!\n";
...
assertEquals(str, new Text(input));
```
You want to make tests as possible, both so they are easy to read and to avoid bugs in tests.